web/src/components/CityEntry.tsx

```diff
@@ -31,50 +30,47 @@ type CityEntryProps = {
   language: string
   city: CityModel
   filterText: string
-  theme: ThemeType
 }
 
-class CityEntry extends React.PureComponent<CityEntryProps> {
-  getMatchedAliases = (city: CityModel, normalizedFilter: string): Array<string> => {
+const CityEntry = ({ language, city, filterText }: CityEntryProps) => {
+  const theme = useTheme()
+  const getMatchedAliases = useCallback((city: CityModel, normalizedFilter: string): Array<string> => {
     if (city.aliases && normalizedFilter.length >= 1) {
       return Object.keys(city.aliases).filter(alias => normalizeString(alias).includes(normalizedFilter))
     }
     return []
-  }
+  }, [])
 
-  render() {
-    const { city, language, filterText, theme } = this.props
-    const normalizedFilter = normalizeString(filterText)
-    const aliases = this.getMatchedAliases(city, normalizedFilter)
-    return (
-      <CityListItem to={cityContentPath({ cityCode: city.code, languageCode: language })}>
-        <Highlighter
-          searchWords={[filterText]}
-          sanitize={normalizeString}
-          aria-label={city.name}
-          textToHighlight={city.name}
-          autoEscape
-          highlightStyle={{ backgroundColor: theme.colors.backgroundColor, fontWeight: 'bold' }}
-        />
-        <div style={{ margin: '0 5px', fontSize: '12px' }}>
-          {aliases.slice(0, MAX_NUMBER_OF_ALIASES).map((alias, index) => (
-            <span key={alias}>
-              <AliasItem
-                aria-label={alias}
-                searchWords={[filterText]}
-                sanitize={normalizeString}
-                textToHighlight={alias}
-                autoEscape
-                highlightStyle={{ backgroundColor: theme.colors.backgroundColor, fontWeight: 'bold' }}
-              />
-              {index !== aliases.length - 1 && <span>, </span>}
-            </span>
-          ))}
-          {aliases.length > MAX_NUMBER_OF_ALIASES && <span> ... </span>}
-        </div>
-      </CityListItem>
-    )
-  }
+  const normalizedFilter = normalizeString(filterText)
+  const aliases = getMatchedAliases(city, normalizedFilter)
```

Comment:
Optional: Probably it would be better to use a `useMemo` here instead of a `useCallback` and then call that: 
``` ts
  const aliases = useMemo((): string[] => {
    const normalizedFilter = normalizeString(filterText)
    if (city.aliases && normalizedFilter.length >= 1) {
      return Object.keys(city.aliases).filter(alias => normalizeString(alias).includes(normalizedFilter))
    }
    return []
  }, [city, filterText])

```